### PR TITLE
scm-record features

### PR DESCRIPTION
**Pull request by arxanas**
Sticky header demo:

https://user-images.githubusercontent.com/454057/236634332-7f8e3853-e7e6-4cb8-894d-7fca52bc66a1.mov

### CODE REVIEW

File: scm-record/src/types.rs [hunk 1]
```diff
@@ -288,6 +288,19 @@ impl File<'_> {
         (acc_selected, acc_unselected)
     }
 
+    /// Set the selection of all sections in this file.
+    pub fn set_toggled(&mut self, toggled: bool) {
```
Comment (martinvonz) [inline scm-record/src/types.rs:356]: What do you think about calling it `set_selected()`? I find "toggled" (the adjective) a bit unclear. I think of "toggle" (the verb) as switching something between two states, but it's not obvious what to me what "set toggled" means. `set_selected()` would also better match the function's description.

File: scm-record/README.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+# Feature wishlist
+
+Here are some features in the UI which are not yet implemented:
+
+- Select inner/outer element.
+- Jump to next/previous element of same kind.
+- Expand/collapse sections.
+- Menu bar to explain available actions and keybindings.
+- Toggle all items within some scope.
+- "Sticky" file and/or section headers.
```
Comment (martinvonz) [inline scm-record/README.md]: Add "edit one side of the diff in an editor"? That's a feature I miss compared to Meld. The editing could be done directly in scm-record, but I suspect we don't want to write our own editor.

File: scm-record/src/ui.rs [hunk 3]
```diff
@@ -211,28 +211,27 @@ impl From<crossterm::event::Event> for Event {
             ) => Self::PageDown,
 
             Event::Key(KeyEvent {
-                code: KeyCode::Up,
+                code: KeyCode::Up | KeyCode::Char('k'),
```
Comment (martinvonz) [inline scm-record/src/ui.rs:215]: Update README with the new keys?

File: scm-record/README.md [hunk 4]
```diff
@@ -57,7 +57,6 @@ Here's some projects that don't use `scm-record`, but could benefit from integra
 
 Here are some features in the UI which are not yet implemented:
 
-- Select inner/outer element.
```
Comment (martinvonz) [inline scm-record/README.md]: Looks like the Keybindings section should be update with `left` and `right` now (and `j` and `l`)

File: scm-record/src/types.rs (hunk 1 above)
Comment (arxanas) [reply scm-record/src/types.rs]: Updated to `checked`. I think a "selection" is an item or subset of items which can then be checked or toggled.

File: scm-record/README.md (hunk 2 above)
Comment (arxanas) [reply scm-record/README.md]: Added.

**Comment by arxanas** [comment]
@martinvonz thanks for the review!

**Review by martinvonz — APPROVED** [review]
Thanks! I use scm-record ~daily

**Pull request merged**